assets/js/translator.js

```diff
@@ -741,6 +764,125 @@ function translateDoc() {
+    // insert noise using document.write that 1. makes things enormously slow, and 2. completely mess up styling so e.g. you
+    // have to scroll through a full screen of whitespace before reaching content. This might mess things up some places – needs
+    // testing – but on the other hand most uses of document.write are evil.
+    return html.replace(/document[.]write[(]/g, 'console.log("document.write "+');
+}
+
+function translateWebpage() {
+    persistChoices('translator', true);
+    if(!$('div#translateWebpage').is(':visible')) {
+        showTranslateWebpageInterface($('#originalText').val().trim());
+    }
+
+    if(pairs[curSrcLang] && pairs[curSrcLang].indexOf(curDstLang) !== -1) {
+        sendEvent('translator', 'translateWebpage', curSrcLang + '-' + curDstLang);
+        if(textTranslateRequest) {
+            textTranslateRequest.abort();
+            clearTimeout(apyRequestTimeout);
+        }
+        $('iframe#translatedWebpage').animate({'opacity': 0.75}, 'fast');
+        textTranslateRequest = callApy({
+            data: {
+                'langpair': curSrcLang + '|' + curDstLang,
+                'markUnknown': 'no',
+                'url': $('input#webpage').val()
+            },
+            dataType: 'json',
+            success: handleTranslateWebpageSuccessResponse,
+            error: handleTranslateWebpageErrorResponse,
+            complete: function () {
+                ajaxComplete();
+                textTranslateRequest = undefined;
+                $('iframe#translatedWebpage').animate({'opacity': 1}, 'fast');
+            }
+        }, '/translatePage', true);
+    }
+}
+
+function handleTranslateWebpageSuccessResponse(data) {
+    if(data.responseStatus === HTTP_OK_CODE) {
+        var iframe = $('<iframe id="translatedWebpage" class="translatedWebpage" frameborder="0"></iframe>')[0];
+        $('#translatedWebpage').replaceWith(iframe);
+        iframe.contentWindow.document.open();
+        var html = cleanPage(data.responseData.translatedText);
+        iframe.contentWindow.document.write(html);
+        iframe.contentWindow.document.close();
+        var contents = $(iframe).contents();
+        contents.find('head')
+            .append($('<base>').attr('href', $('input#webpage').val()));
+        $(iframe).load(function () {
+            $.each(contents.find('a'), function (index, a) {
+                var href = a.href;
+                $(a).on('click', function () { window.parent.translateLink(href); });
+                a.href = '#';
+                a.target = '';
+            });
+        });
+    }
+    else {
+        translationNotAvailableWebpage(data);
+    }
+}
+
+function handleTranslateWebpageErrorResponse(jqXHR) {
+    translationNotAvailableWebpage(jqXHR.responseJSON);
+}
+
+function showTranslateWebpageInterface(url) {
+    $('#srcLangSelectors').css({
```

Comment:
These numbers correspond to max `width` available for the src languages. Some padding is added to it in the event Cancel button appears for better UI :P . I'll add a class and code these styles in that class in the CSS file. The `calc` value corresponds to one of the fixes I had made with respect to the overlapping of src languages and swap button. And hence, the values.